### PR TITLE
Implement query for active emissions

### PR DESCRIPTION
**Pull request by pacmanifold**

### CODE REVIEW

File: contracts/incentives/src/contract.rs [hunk 1]
```diff
@@ -557,9 +558,39 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
...
+                &collateral_denom,
+                &incentive_denom,
+                env.block.time.seconds(),
+            )?;
+
+            Ok(emission.map(|e| (incentive_denom, e)))
+        })
+        .collect::<StdResult<Vec<_>>>()?
+        .into_iter()
+        .filter_map(|x| x)
```
Comment (piobab) [inline contracts/incentives/src/contract.rs:588]: `flatten()`

File: contracts/incentives/tests/test_quering.rs [hunk 2]
```diff
@@ -223,3 +225,63 @@ fn query_emissions() {
...
+    // Setup emissions
+    EMISSIONS.save(deps.as_mut().storage, ("uusdc", "uosmo", 604800), &Uint128::new(100)).unwrap();
+    EMISSIONS
+        .save(deps.as_mut().storage, ("uusdc", "umars", 604800 * 2), &Uint128::new(50))
+        .unwrap();
+    EMISSIONS
+        .save(deps.as_mut().storage, ("uusdc", "uosmo", 604800 * 2), &Uint128::new(100))
+        .unwrap();
+
+    // Query before emission start
```
Comment (piobab) [inline contracts/incentives/tests/test_quering.rs:276]: to be removed?

**Review by piobab — APPROVED** [review]

File: contracts/incentives/src/contract.rs (hunk 1 above)
Comment (pacmanifold) [reply contracts/incentives/src/contract.rs]: fixed

File: contracts/incentives/tests/test_quering.rs (hunk 2 above)
Comment (pacmanifold) [reply contracts/incentives/tests/test_quering.rs]: removed comment

**Review by piobab — APPROVED** [review]

**Pull request merged**